shell.nix

```diff
@@ -24,24 +24,24 @@ let
 
     # These programs will be available inside the nix-shell.
     buildInputs = with haskellPackages; [
-      #cardanoDbSync.cardano-db-sync
-      #cardanoNode.cardano-cli
-      #cardanoNode.cardano-node
-      #cabal-install
-      #stack
-      #ghcid
-      #hlint
-      stylish-haskell
-      #weeder
-      #gnumeric
-      #libreoffice
+      # cardanoDbSync.cardano-db-sync
+      cardanoNode.cardano-cli
+      cardanoNode.cardano-node
```

Comment:
cardano-cli and cardano-node should not be built here. these are built when needed using the call to `prebuild`, e.g. in shelley3pools/benchmark.sh

Reply:
That suggestion breaks the Nix+Cabal workflow:

```
cabal: Internal error in target matching. It should always be possible to find
a syntax that's sufficiently qualified to give an unambiguous match. However
when matching 'cardano-node:exe:cardano-node' we found
cardano-node:exe:cardano-node (unknown-component) which does not have an
unambiguous syntax. The possible syntax and the targets they match are as
follows:
'cardano-node:exe:cardano-node' which matches cardano-node:exe:cardano-node
(unknown-component), cardano-node:exe:cardano-node (unknown-file)
':pkg:cardano-node:exe:cardano-node' which matches
cardano-node:exe:cardano-node (unknown-component)
```

Reply:
would it be possible to have a specific .nix file for your type of workflow?  `nix-shell prebuilt.nix`

Reply:
or, the other way around: `nix-shell minimal.nix` ?

Reply:
@CodiePP, done -- implemented the `shell-minimal.nix` approach.